### PR TITLE
Fix #3756: Test that persistent licenses cannot be ingested into temp…

### LINKED ISSUE #3756: [eme] Add tests for rejecting improper licenses for the session type
The [`update()` algorithm](https://w3c.github.io/encrypted-media/#update) says the CDM should "Reject promise with a newly created TypeError" if a persistable license is provided to a session type other than `"persistent-license"` and vice versa.

The drmtoday server appears to request a different license based on a custom header (see `getCustomHeaders()` in `drm-messagehandler.js`), so we should be able to test these negative cases.

Note that failing this test could cause a key to be unexpectedly persisted, which could hide problems in other tests. It might be good to have a license with a key that does not appear in other tests.

### PR DESCRIPTION
**Pull request by mwatson2**
…orary sessions

### CODE REVIEW

File: encrypted-media/drm-temporary-license-type.html [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    <div id='log'></div>
+
+    <script>
+        var keysystem = getSupportedKeySystem();
+
+        getSupportedContentAndInitDataTypes( keysystem )
+        .then( function( supported )
+        {
+            // This test requires initData for supported content.
+            // We prefer explciit initData configured for the content, but
```
Comment (AlexeyKor) [inline encrypted-media/drm-temporary-license-type.html]: nit: explciit -> explicit

File: encrypted-media/drm-temporary-license-type.html [hunk 2]
```diff
@@ -0,0 +1,71 @@
+<!doctype html>
+<html>
+  <head>
+    <meta charset=utf-8>
+    <title>Encrypted Media Extensions: Events with DRM</title>
```
Comment (AlexeyKor) [inline encrypted-media/drm-temporary-license-type.html]: Title doesn't reflect the content of the test.

File: encrypted-media/scripts/temporary-license-type.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+            assert_true(event instanceof window.MediaKeyMessageEvent);
+            assert_equals(event.target, mediaKeySession);
+            assert_equals(event.type, 'message');
+            assert_any( assert_equals,
+                        event.messageType,
+                        [ 'license-request', 'individualization-request' ] );
+
+            config.messagehandler( event.messageType, event.message ).then( function( response ) {
+                mediaKeySession.update( response ).then( test.step_func( function() {
+                    if ( event.messageType !== 'license-request' )
```
Comment (AlexeyKor) [inline encrypted-media/scripts/temporary-license-type.js]: nit: Add { }

**Review by AlexeyKor — APPROVED** [review]
LGTM

**Comment by jdsmith3000** [comment]
LGTM as well.

**Pull request merged**